consensus/src/lib.rs

```diff
@@ -160,29 +162,27 @@ impl<N: GossipNetwork, S: Storage> Consensus<N, S> {
         round_zero_timestamp: Timestamp,
         this_node_key: Option<PrivateKey>,
     ) -> Result<Self, Error> {
-        let state = state_storage.read_file(STATE_FILE_NAME).await?;
-        let state: State = serde_json::from_str(&state)?;
-        // TODO: check if `this_node_key` is in the validator set. If not, error.
-        if block_header != state.block_header {
-            let height_info = generate_height_info(
-                &block_header,
-                consensus_parameters,
-                round_zero_timestamp,
-                this_node_key.clone(),
-            )?;
+        // Do a pre-calculation for prettier code.
```

Comment:
How about `Prepare new state for the possible storage reset case`

Reply:
applied